src/load/azext_load/data_plane/load_test_run/custom.py

```diff
@@ -0,0 +1,539 @@
+    except ResourceNotFoundError as e:
+        msg = f"Test run with given test run ID : {test_run_id} does not exist."
+        logger.debug(msg)
+        raise InvalidArgumentValueError(msg) from e
+    test_run_body = create_or_update_test_run_body(
+        test_run_body.get("testId"), description=description
+    )
+    logger.info("Updating test run %s", test_run_id)
+    # pylint: disable-next=protected-access
+    response = client._test_run_initial(test_run_id=test_run_id, body=test_run_body)
+    logger.debug("Test run updated with following response %s", response)
+    logger.info("Update test run completed")
+    return response
+
+
+def delete_test_run(cmd, load_test_resource, test_run_id, resource_group_name=None):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    logger.info("Deleting test run %s", test_run_id)
+    response = client.delete_test_run(test_run_id=test_run_id)
+    logger.debug("Test run deleted with following response %s", response)
+    logger.info("Delete test run completed")
+    return response
+
+
+def list_test_runs(cmd, test_id, load_test_resource, resource_group_name=None):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    logger.info("Listing test runs for test %s", test_id)
+    response = client.list_test_runs(test_id=test_id)
+    logger.debug("Test runs listed with following response %s", response)
+    logger.info("List test runs completed")
+    return response
+
+
+def stop_test_run(cmd, load_test_resource, test_run_id, resource_group_name=None):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    logger.info("Stopping test run %s", test_run_id)
+    response = client.stop_test_run(test_run_id=test_run_id)
+    logger.debug("Test run stopped with following response %s", response)
+    logger.info("Stop test run completed")
+    return response
+
+
+def download_test_run_files(
+    cmd,
+    load_test_resource,
+    test_run_id,
+    path,
+    test_run_input=False,
+    test_run_log=False,
+    test_run_results=False,
+    resource_group_name=None,
+    force=False,  # pylint: disable=unused-argument
+):
+    logger.info("Downloading test run files for test run %s", test_run_id)
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test_run_data = client.get_test_run(test_run_id=test_run_id)
+    if test_run_data.get("testArtifacts") is None:
+        logger.warning("No test artifacts found for test run %s", test_run_id)
+
+    if test_run_input:
+        logger.info("Downloading input artifacts for test run %s", test_run_id)
+        if test_run_data.get("testArtifacts", {}).get("inputArtifacts") is not None:
+            input_artifacts = test_run_data.get("testArtifacts", {}).get(
+                "inputArtifacts", {}
+            )
+            files_to_download = []
```

Comment:
nit: move `files_to_download` outside of all ifs for which file to download
Given if, add the file to `files_to_download` 
Then in the end, if its not empty, download the files

Reply:
We want to give separate destination paths for different download option (input artifacts, logs files and result files). And multiple files will be only in case of input artifacts hence this is particularly inside input artifact option.

Reply:
That makes sense.